music/src/ddsp/model.ts

```diff
@@ -0,0 +1,110 @@
+/**
+ * Core implementation for [DDSP]{@link
+ * https://g.co/magenta/ddsp} models.
+ *
+ * @license
+ * Copyright 2020 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Imports.
+ */
+import * as tf from '@tensorflow/tfjs';
+import '@tensorflow/tfjs-backend-webgl';
+
+import { getAudioFeatures, toneTransfer, memCheck } from './ddsp';
+import { MODEL } from './constants';
+import { AudioFeatures } from './interfaces';
+
+class DDSP {
+  private initialized: boolean;
+
+  /**
+   * `DDSP` constructor.
+   *
+   * @param model The model to use
+   */
+  constructor() {
```

Comment:
If we're following the model API for the rest of magenta.js I think it makes sense to provide the constructor with the url and constants, for example: https://github.com/magenta/magenta-js/blob/master/music/src/gansynth/model.ts#L30